### PR TITLE
#4 :sparkles: support deactivating a migration by name; improve examp…

### PR DESCRIPTION
**Pull request by d-ryan-ashcraft**
…le projects

### CODE REVIEW

File: baton-maven-plugin/src/main/java/org/technologybrewery/baton/BatonMojo.java [hunk 1]
```diff
@@ -132,23 +136,40 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
-                executionSummary.addMigrationSummary(migrationSummary);
-
-            } catch (Exception e) {
-                throw new BatonException("Could not complete migrations!", e);
+            if (isActive(target)) {
+                try {
+                    getLog().debug(String.format("Executing Migration: %s (%s)", target.getName(), target.getImplementation()));
+                    Class<Migration> implementationClass = (Class<Migration>) Class.forName(target.getImplementation());
+                    Constructor<Migration> constructor = implementationClass.getConstructor();
+                    Migration migration = constructor.newInstance();
```
Comment (d-ryan-ashcraft) [inline baton-maven-plugin/src/main/java/org/technologybrewery/baton/BatonMojo.java:144]: I: realized that name and description were never naming it to the Migration instance

File: baton-maven-plugin/src/main/java/org/technologybrewery/baton/BatonMojo.java [hunk 2]
```diff
@@ -231,6 +252,6 @@ protected FileSet[] getFileSetsForTarget(MigrationTarget target) {
             localFileSets.add(fileSet);
         }
```
Comment (d-ryan-ashcraft) [inline baton-maven-plugin/src/main/java/org/technologybrewery/baton/BatonMojo.java:254]: I: edge case found while unit testing; won't happen in real use, but makes this more predictable when you hack it in tests

File: baton-maven-plugin/src/test/java/org/technologybrewery/baton/BatonMojoTestSteps.java [hunk 3]
```diff
@@ -11,27 +11,44 @@
 
 import java.io.File;
 import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
-public class MigrationTargetSteps {
```
Comment (d-ryan-ashcraft) [inline baton-maven-plugin/src/test/java/org/technologybrewery/baton/BatonMojoTestSteps.java:16]: I: improving naming for better clarity

File: examples/example-migration-configuration/src/main/java/org/technologybrewery/baton/LoggingUnsuccessfulMigration.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        return true;
+    }
+
+    @Override
+    protected boolean performMigration(File file) {
+        logger.info("Performing migration `{}` on file `{}`", getName(), file.getAbsoluteFile());
+        return true;
```
Comment (d-ryan-ashcraft) [inline examples/example-migration-configuration/src/main/java/org/technologybrewery/baton/LoggingUnsuccessfulMigration.java]: S: should return false so it is unsuccessful, as the class name denotes...

File: examples/example-migration-configuration/src/main/resources/migrations.json [hunk 5]
```diff
@@ -1,11 +1,19 @@
 [
   {
```
Comment (d-ryan-ashcraft) [inline examples/example-migration-configuration/src/main/resources/migrations.json:2]: I: removed Toml stuff since that doesn't really help anyone.  Updates example projects to be simple instantiations of things users might really want to know how to do instead

File: examples/example-projects/ignore-migration-example/pom.xml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+    <artifactId>ignore-migration-example</artifactId>
+    <packaging>jar</packaging>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>${project.groupId}</groupId>
+                <artifactId>baton-maven-plugin</artifactId>
+                <configuration>
```
Comment (d-ryan-ashcraft) [inline examples/example-projects/ignore-migration-example/pom.xml:21]: I: here is the configuration needed to deactivate a migration

File: baton-maven-plugin/src/main/java/org/technologybrewery/baton/AbstractMigration.java [hunk 7]
```diff
@@ -51,8 +51,12 @@ public MigrationSummary execute(FileSet[] fileSets) {
                 File fileToMigrate = new File(fileSet.getDirectory(), file);
                 if (shouldExecuteOnFile(fileToMigrate)) {
                     try {
-                        performMigration(fileToMigrate);
```
Comment (d-ryan-ashcraft) [inline baton-maven-plugin/src/main/java/org/technologybrewery/baton/AbstractMigration.java:54]: I (Informational): fixed this defect where migrations were reported as successes unless they threw an exception

**Review by aaron-gary — APPROVED** [review]
